truss-chains/truss_chains/deploy.py

```diff
@@ -2,10 +2,23 @@
 import inspect
 import logging
 import pathlib
-from typing import Any, Dict, Iterable, Iterator, MutableMapping, Optional, Type, cast
+import random
+import string
+from typing import (
+    Any,
+    Dict,
+    Iterable,
+    Iterator,
+    List,
+    MutableMapping,
+    Optional,
+    Type,
+    cast,
+)
 
 import truss
 from truss.remote.baseten import service as b10_service
+from truss.remote.baseten.types import ChainletData
```

Comment:
Nit/Optional: for consistency with importing modules and not symbols[1] you could import this module and alias it (because "types" should not be in the module namespace):

```
from truss.remote.baseten import types as b10_types
...
b10_types.ChainletData
```

[1] This is follwing Google Python style guide which I read was supposed to be used at baseten, even though it isn't followed much, but I tried to be consistent within the chains sub-tree.